tcms/testruns/views.py

```diff
@@ -122,15 +122,19 @@ def post(self, request):
         return render(request, self.template_name, context_data)
 
 
-@require_GET
-def search(request):  # pylint: disable=missing-permission-required
-    form = SearchRunForm(request.GET)
-    form.populate(product_id=request.GET.get('product'))
+class SearchTestRunView(TemplateView):  # pylint: disable=missing-permission-required
 
-    context_data = {
-        'form': form,
-    }
-    return render(request, 'testruns/search.html', context_data)
+    template_name = 'testruns/search.html'
+
+    def get_context_data(self, **kwargs):
+        form = SearchRunForm(self.request.GET)
+        form.populate(product_id=self.request.GET.get('product'))
+
+        context_data = {
+            'form': form,
+        }
```

Comment:
You can `return` directly the dict literal here and save a few lines of code/1 unneeded variable. Up to you.